src/spark/utils.py

```diff
@@ -19,14 +22,39 @@ def get_spark_session(app_name: str, local: bool = False) -> SparkSession:
 
     spark_conf = SparkConf()
 
-    spark_conf.setAll(
-        [
-            (
-                "spark.master",
-                os.environ.get("SPARK_MASTER_URL", "spark://spark-master:7077"),
-            ),
-            ("spark.app.name", app_name),
-        ]
-    )
+    if delta_lake:
+
+        jars_dir = "/opt/bitnami/spark/jars/"
+        jar_files = [os.path.join(jars_dir, f) for f in os.listdir(jars_dir) if f.endswith(".jar")]
+        jars = ",".join(jar_files)
+
+        spark_conf.setAll(
+            [
+                (
+                    "spark.master",
+                    os.environ.get("SPARK_MASTER_URL", "spark://spark-master:7077"),
+                ),
+                ("spark.app.name", app_name),
+                ("spark.hadoop.fs.s3a.endpoint", os.environ.get("MINIO_URL")),
+                ("spark.hadoop.fs.s3a.access.key", os.environ.get("MINIO_ACCESS_KEY")),
+                ("spark.hadoop.fs.s3a.secret.key", os.environ.get("MINIO_SECRET_KEY")),
+                ("spark.jars", jars),
+                ("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension"),
+                ("spark.sql.catalog.spark_catalog", "org.apache.spark.sql.delta.catalog.DeltaCatalog"),
+                ("spark.hadoop.fs.s3a.path.style.access", "true"),
+                ("spark.hadoop.fs.s3a.connection.ssl.enabled", "false"),
```

Comment:
This should probably be a parameter so we don't accidentally make everything insecure later

Reply:
I just removed this and leave it as default. I think it doesn't matter here.

Reply:
:+1: